rasa/core/policies/ensemble.py

```diff
@@ -91,590 +218,315 @@ def _check_for_important_policies(self) -> None:
-                    f"Module for policy '{policy_name}' could not "
-                    f"be loaded. Please make sure the "
-                    f"name is a valid policy."
-                ) from e
-
-        return parsed_policies
-
-    @classmethod
-    def get_featurizer_from_dict(cls, policy: Dict[Text, Any]) -> Tuple[Any, Any]:
-        """Gets the featurizer initializer and its arguments from a policy config."""
-        # policy can have only 1 featurizer
-        if len(policy["featurizer"]) > 1:
-            raise InvalidPolicyConfig(
-                f"Every policy can only have 1 featurizer "
-                f"but '{policy.get('name')}' "
-                f"uses {len(policy['featurizer'])} featurizers."
-            )
-        featurizer_config = policy["featurizer"][0]
-        featurizer_name = featurizer_config.pop("name")
-        featurizer_func = registry.featurizer_from_module_path(featurizer_name)
-
-        return featurizer_func, featurizer_config
+        config: Dict[Text, Any],
+        model_storage: ModelStorage,
+        resource: Resource,
+        execution_context: ExecutionContext,
+    ) -> GraphComponent:
+        """Creates a new instance (see parent class for full docstring)."""
+        return cls()
 
-    @classmethod
-    def get_state_featurizer_from_dict(
-        cls, featurizer_config: Dict[Text, Any]
-    ) -> Tuple[Callable, Dict[Text, Any]]:
-        # featurizer can have only 1 state featurizer
-        if len(featurizer_config["state_featurizer"]) > 1:
-            raise InvalidPolicyConfig(
-                f"Every featurizer can only have 1 state "
-                f"featurizer but one of the featurizers uses "
-                f"{len(featurizer_config['state_featurizer'])}."
-            )
-        state_featurizer_config = featurizer_config["state_featurizer"][0]
-        state_featurizer_name = state_featurizer_config.pop("name")
-        state_featurizer_func = registry.state_featurizer_from_module_path(
-            state_featurizer_name
-        )
+    def __str__(self) -> Text:
+        return f"{self.__class__.__name__}()"
 
-        return state_featurizer_func, state_featurizer_config
+    def validate(self, ensemble: List[Policy], domain: Domain) -> None:
+        """Checks whether this ensemble can be applied to the given policies.
 
-
-class SimplePolicyEnsemble(PolicyEnsemble):
-    """Default implementation of a `Policy` ensemble."""
+        Args:
+          ensemble: a list of policies
+          domain: the common domain
+        Raises:
+          `InvalidPolicyEnsembleConfig`: if this ensemble cannot be applied to
+            the given ensemble of policies
+        """
+        super().validate(ensemble=ensemble, domain=domain)
+        # TODO: shouldn't we raise in case priorities are not unique since this might
+        # lead to unexpected results in edge cases (swap order in your config)
+        self.warn_priorities_not_unique(ensemble=ensemble)
```

Comment:
Even then there's the case where they're equally confident and a tie breaker is needed - it makes sense to use priorities for that. If the order of policy predictions (wrt policies who made it) never changes, that order could be used as tie breaker but that seems to inexplicit (and would fail in that very strange case, where you move things around in your config and assume that has no effect).

Reply:
In principle, we shouldn't compare confidences of two distinct ML policies in the first place. 🙈

Reply:
I think for now we should just emulate exactly whatever we have in Rasa 2.8.x.

Reply:
>In principle, we shouldn't compare confidences of two distinct ML policies in the first place. 🙈

And I still think we shouldn't call these things confidences 😅   - but I'll revert the probability stuff because Daksh just reminded me that it's not even consistently discrete probability distributions  but can be an array of probabilities .... 😬